rfd/0128-aws-integration-tests.md

```diff
@@ -0,0 +1,191 @@
+and how they will be authorized to perform the required actions.
+
+Teleport AWS account will be configured to allow GitHub OIDC provider to assume
+a set of roles. These roles will be assumed by the GitHub actions pipeline to
+interact with AWS API.
+
+GitHub action [configure-aws-credentials](https://github.com/aws-actions/configure-aws-credentials)
+will be used to handle the authentication with AWS API. This action configures
+AWS credentials and region environment variables for use in subsequent steps.
+The action implements the AWS SDK credential resolution chain and exports the
+following environment variables:
+
+- `AWS_ACCESS_KEY_ID`
+- `AWS_SECRET_ACCESS_KEY`
+- `AWS_SESSION_TOKEN`
+
+Once the environment variables are set, the AWS SDK that Teleport uses will
+automatically use them to authenticate with the AWS API and does not require
+any additional configuration.
+
+One of the requirements for the integration tests is that they should use the
+minimum required permissions to perform the required actions. This will ensure
+that we don't introduce regressions by changing the permissions required by
+Teleport. To achieve this, GitHub action will be allowed to assume a set of
+roles that will simulate the minimum required permissions that each Teleport
+service requires to perform the required actions. Teleport services will be
+configured to assume these roles when running the integration tests - requires
+changes to the Teleport configuration - so we can test them in isolation.
+
+The AWS account will be configured with a set of roles that will be assumed by
+the GitHub actions pipeline. These roles will be configured with the minimum
+required permissions to run the integration tests. This will ensure that we
+don't introduce regressions by changing the permissions required by Teleport
+but those changes are not detected because another test requires the same
+permissions.
+
+AWS account configuration will be handled by IAC and will be stored in the
+Teleport Terraform repository. This will allow us to track changes to the configuration
+and ensure that we can revert them if needed while also allowing us to review
+the changes before they are applied. Each End-to-End test will run in a separate
+Job so we can configure the assumed role for each test. End-to-End action
+requires `id-token: write` permission to the GitHub OIDC provider to generate
+the OIDC token that will be used to authenticate with AWS API.
+
+## Tests
+
+This section describes the integration tests that will be added to the Teleport
+and what they will test.
+
+### Kubernetes Access
+
+Teleport supports the automatic discovery of Kubernetes clusters running on AWS.
+This is done by using the AWS API to discover the Kubernetes clusters running on
+the account that matches the configured labels and then using the Kubernetes API
+to forward the requests to the cluster.
+
+The first step of this process happens on the Teleport Discovery service and
```

Comment:
Can you refactor this section to be more specific please w/r/t what kind of Teleport services will be provisioned and what EKS clusters it will be discovering. Basically, instead of explaining internals (i.e. kube_cluster objects, etc.) I would focus on explaining test scenarios. For example:

1. The test will start discovery_service with config "xxx" and kube_service with config "yyy".
2. Discovery service will look for EKS clusters with tags "a: b".
3. The test will expect the EKS cluster "a: b" to join and will execute "kubectl get pods" on it.

And so on.

Reply:
Updated in [9d98ad5](https://github.com/gravitational/teleport/pull/27156/commits/9d98ad553b6f2aadb3eb695f5d549d73ea21f8c4)